src/path/sentry_path_windows.c

```diff
@@ -125,7 +124,13 @@ sentry__path_dir(const sentry_path_t *path)
     if (!dir_path) {
         return NULL;
     }
-    PathRemoveFileSpecW(dir_path->path);
+
+    // find the filename part and truncate just in front of it if possible
+    sentry_pathchar_t *filename
+        = (sentry_pathchar_t *)sentry__path_filename(dir_path);
+    if (filename > dir_path->path) {
+        *(filename - 1) = L'\0';
```

Comment:
C is so scary

Reply:
Hahaha, I constantly facepalming why we have to write all this horrible code over and over again just because C does not have a sufficiently good std lib, such as Rusts `Path(Buf)` / `&str/String` 🤦‍♂️